CodenameOne/src/com/codename1/ui/Image.java

```diff
@@ -1207,4 +1208,347 @@ public void fireChangedEvent() {
+     * from the repository
+     * <a href="https://github.com/recurser/exif-orientation-examples">EXIF
+     * Orientation-flag example images</a></p>
+     * <p>
+     * Code example:</p>
+     * <script src="https://gist.github.com/jsfan3/7fc101523955e8179fadd2c713a09e05.js"></script>
+     *
+     * @param capturedImage is the FileSystemStorage path of a captured photo,
+     * usually inside a temporary directory
+     * @param rotatedImage is the FileSystemStorage path in which the capture
+     * photo is stored, normally this should be inside the
+     * FileSystemStorage.getAppHomePath().
+     * @return the rotated and/or flipped image
+     */
+    public static Image exifRotation(String capturedImage, String rotatedImage) throws IOException {
+        return exifRotation(capturedImage, rotatedImage, -1);
+    }
+
+    /**
+     * <p>
+     * The main use case of this method is the automatic rotation and flipping
+     * of an image returned from the camera or from the gallery, preserving the
+     * original format (jpeg or png); it detects the Exif Orientation Tag, if
+     * available (all the possible Exif Orientation Tag values are
+     * supported).</p>
+     * <p>
+     * However, rotating and/or flipping an hi-res image is very inefficient,
+     * that's why you should consider to pass a maxSize value as small as
+     * possible: it makes this method working faster.</p>
+     * <p>
+     * If there is no rotation or flipping, or if the capturedImage has a format
+     * different from jpeg and png, the image is only copied or scaled if
+     * necessary.<br>Note that this method doesn't rely on the file extension,
+     * but on the mime type of the capturedImage, since some devices don't give
+     * appropriate extension to images returned from the gallery.</p>
+     * <p>
+     * You can test all the possible orientation values downloading the images
+     * from the repository
+     * <a href="https://github.com/recurser/exif-orientation-examples">EXIF
+     * Orientation-flag example images</a></p>
+     * <p>
+     * Code example:</p>
+     * <script src="https://gist.github.com/jsfan3/7fc101523955e8179fadd2c713a09e05.js"></script>
+     *
+     * @param capturedImage is the FileSystemStorage path of a captured photo,
+     * usually inside a temporary directory
+     * @param rotatedImage is the FileSystemStorage path in which the capture
+     * photo is stored, normally this should be inside the
+     * FileSystemStorage.getAppHomePath().
+     * @param maxSize is the maximum value of the width and height of the
+     * rotated images, that is scaled if necessary, keeping the ratio.
+     * @return the com.codename1.ui.Image
+     * @throws java.io.IOException
+     */
+    public static Image exifRotation(String capturedImage, String rotatedImage, int maxSize) throws IOException {
+        FileSystemStorage fss = FileSystemStorage.getInstance();
```

Comment:
This API currently only works with file system storage. It would be great if it can work with a header byte array.